### ISSUE TITLE
Training object detector on lidar

### ISSUE THREAD

**Issue by connorlee77** [princeton-computational-imaging/SeeingThroughFog#45]
Has anyone successfully trained an object detector (yolov3, ssd, etc...) on just lidar_hdl64_last_stereo_left or lidar_hdl64_strong_stereo_left? I'm getting pretty trash results, which I believe is due to the sparsity of the points. Anyone got tips for training?

**Comment by MartinHahner** [comment]
Yes, I was able to train models using the [OpenPCDet](https://github.com/open-mmlab/OpenPCDet) repository.
I converted the [KittiDataset class](https://github.com/open-mmlab/OpenPCDet/blob/master/pcdet/datasets/kitti/kitti_dataset.py) and used [this calib.txt](https://github.com/princeton-computational-imaging/SeeingThroughFog/files/6513821/calib.txt).

I currently have a submission under review, so I am not willing to share my code yet, 
but if it gets accepted, we plan to make the code public.

Hope this helps for now.

**Comment by connorlee77** [comment]
Interesting. Just to clarify, you were able to train good models using just the intensity data specifically from lidar_hdl64_last_**stereo_left** rather than the 3d point clouds in lidar_hdl64_last?

**Comment by MartinHahner** [comment]
I used `lidar_hdl64_strongest` (and `lidar_hdl64_last`).

**Comment by barzanisar** [comment]
Hey @MartinHahner can you please share your modified kitti_dataset.py, pv_rcnn.yaml/pointpillars.yaml and dataset config files for seeing through fog? Thanks!

**Comment by kyrie-louy** [comment]
Hey @MartinHahner, could you please share your modified kitti_dataset.py, model/dataset config files for the SeeingThroughFog dataset if possible? Many thanks!

> Yes, I was able to train models using the [OpenPCDet](https://github.com/open-mmlab/OpenPCDet) repository. I converted the [KittiDataset class](https://github.com/open-mmlab/OpenPCDet/blob/master/pcdet/datasets/kitti/kitti_dataset.py) and used [this calib.txt](https://github.com/princeton-computational-imaging/SeeingThroughFog/files/6513821/calib.txt).
> 
> I currently have a submission under review, so I am not willing to share my code yet, but if it gets accepted, we plan to make the code public.
> 
> Hope this helps for now.

**Comment by MartinHahner** [comment]
@barzanisar Do you have a clean version that you can share?

**Comment by barzanisar** [comment]
Here is the clean version: https://github.com/barzanisar/OpenPCDet/tree/STF_cc
I only trained with pv_rcnn.yaml
Other model cfg files need to have matching class names: ['PassengerCar', 'Pedestrian', 'RidableVehicle']

**Comment by kyrie-louy** [comment]
> Here is the clean version: https://github.com/barzanisar/OpenPCDet/tree/STF_cc I only trained with pv_rcnn.yaml Other model cfg files need to have matching class names: ['PassengerCar', 'Pedestrian', 'RidableVehicle']

Many thanks for your replies @barzanisar @MartinHahner. It helps a lot!

**Comment by hy3121** [comment]
Should this programme run on Ubuntu System?

**Comment by iyi-alam** [comment]
@MartinHahner @barzanisar Do you have any saved checkpoints of pv_rcnn.yaml config model that you have trained on STF dataset? I am trying to reproduce the results of the paper but unable to get till that point.

**Comment by MartinHahner** [comment]
I don't have any checkpoints anymore.

**Comment by iyi-alam** [comment]
Hi @MartinHahner @barzanisar  Sir,
Did you use union of lidar strongest return and lidar last return as your point cloud? I mean how did you combine both sensor data? I am not getting detection results shown in paper, hence checking if there is a gap in understanding.

Thanks.

**Comment by MartinHahner** [comment]
No, I did not use the union, I either used the `strongest` or the `last` return, but not both together.
